src/subgraph/static-reshape.c

```diff
@@ -86,6 +89,10 @@ static enum xnn_status reshape_copy_operator(
     default:
       XNN_UNREACHABLE;
   }
+  if (status != xnn_status_success) {
+    return status;
+  }
+  return reshape_unary_elementwise_op(opdata, values, num_values, old_workspace_size, threadpool);
```

Comment:
IIUC, we might want to restrict this to a handful of special cases like squeeze (i.e. [a,b,1] --> [a,b], or [a,1,b] --> [a,b]) or unsqueeze (i.e. [a,b] --> [a,b,1] or [a,1,b]) otherwise this can get tricky for the general case.

Reply:
static-reshape can no longer reshape output. This is because the output shape must be static so can never change.

Reply:
I agree with the high level argument. However, squeeze/unsqueeze or adding/removing a dim with size 1, or decreasing a rank without changing strides is a common use for this op (at least in PyTorch), and excluding this may restrict the number of models which we can support with dynamic shape artificially IMHO. For now, perhaps can we put tight constraints on what is allowed to be on the safe side, WDYT?

Reply:
We can loosen the constraint on the new shape being static, however, it must be known when `reshape` is called. Perhaps static-reshape is not the right name anymore. This should be handled in a separate PR.

Reply:
Yeah. Sounds good. I guess this particular issue of capturing user's intent. That is if users want us to change the output shape or not. Like `resize` is another. Many times I have seen users using `resize` to normalize the output image shape regardless of the input image size.

So I guess we can do two things (for those ops),
(1) change the name to static_op --> op
(2) add a flag to allow the op to keep the output shape static

Alternatively, we can keep both static_op and add a new op but only at the Subgraph level.